### ISSUE TITLE
2.6 backport PR request list

### ISSUE THREAD

**Issue by tengyifei** [pytorch/xla#8455]
This is a tracker for backport/cherry-picks into 2.6. For any PRs you want to backport to 2.6, please reply with following:

- Original PR link (this PR should merge into `master`)
- Reason to backport
- 2.6 backport PR link (a separate PR should be created, and that PR should merge into `r2.6`)

This process is similar to the backport request thread in 2.5 release https://github.com/pytorch/xla/issues/7977

Please note the criterion for cherrypicking into 2.6: https://github.com/pytorch/xla/issues/7203

**Comment by tengyifei** [comment]
@lsy323 libtpu pin update to 0.0.6: https://github.com/pytorch/xla/pull/8480.
Reason: pick stable libtpu for Trillium
Backport PR link: @tengyifei will upload one later

**Comment by yaochengji** [comment]
Fix a DDP graph capture issue https://github.com/pytorch/xla/pull/8489
Reason: The DDP result before this patch is wrong
Backport PR link: https://github.com/pytorch/xla/pull/8500

**Comment by mcuiaws** [comment]
Original PR: Compute and hash buffer_donor_indices for step marker #8467
Reason: Fixes tensor corruption issue
Backport PR: #8503

**Comment by mcuiaws** [comment]
Original PR: xm.save() should not set sync_xla_data=True when sync'ing. #8484
Reason: Fixes tensor corruption issues, easily reproducible by running huggingface tutorials.
Backport PR: #8504

**Comment by mcuiaws** [comment]
Original PR: Add xm.xla_device_kind() to return XLA device kind string. #8493
Reason: Requested by Neuron customers. Feature available in JAX but not PyTorch/XLA. Should be very low risk.
Backport PR: #8506

**Comment by jeffhataws** [comment]
Original PR:  When modifying IR node, make sure to not lose the read_only bit https://github.com/pytorch/xla/pull/8505
Reason: Fixed a bug where 0-dimensional tensors result in aliasing errors
Backport PR: TBD

**Comment by mcuiaws** [comment]
> Original PR: When modifying IR node, make sure to not lose the read_only bit #8505 Reason: Fixed a bug where 0-dimensional tensors result in aliasing errors Backport PR: TBD

Backport PR: #8508

**Comment by avizon-aws** [comment]
Cherry-Pick PR for softmax autocast: https://github.com/pytorch/xla/pull/8511

**Comment by savitha-aws** [comment]
Originial PR: Add xla autocast support, update autocast APIs in checkpointing [#8523](https://github.com/pytorch/xla/pull/8523)
Reason: PR adds missing XLA autocast support in gradient checkpointing and updates deprecated APIs for cuda and cpu autocast.
Backport PR: [#8527](https://github.com/pytorch/xla/pull/8527)

**Comment by rpsilva-aws** [comment]
Original PR: Introduce deterministic hash for user computations https://github.com/pytorch/xla/pull/8539
Original issue: https://github.com/pytorch/xla/issues/8537
Reason: Fixes a day-one bug with the user computation hash
Backport PR: https://github.com/pytorch/xla/pull/8554

**Comment by rpsilva-aws** [comment]
Original PR: Metadata agnostic hash for user computations https://github.com/pytorch/xla/pull/8550
Original feature: https://github.com/pytorch/xla/issues/8538
Reason: User computation is cache agnostic to OpMetadata in the HLO module proto, not influencing execution semantics of the computation.
Backport PR: TBD (dependency on https://github.com/pytorch/xla/pull/8554)

**Comment by rpsilva-aws** [comment]
> Original PR: Metadata agnostic hash for user computations #8550 Original feature: #8538 Reason: User computation is cache agnostic to OpMetadata in the HLO module proto, not influencing execution semantics of the computation. Backport PR: TBD (dependency on #8554)

Backport PR: https://github.com/pytorch/xla/pull/8557

**Comment by tengyifei** [comment]
Original PR: https://github.com/pytorch/xla/pull/8521
Reason: prerequisite to fix DDP tests in CI
Backport PR link: https://github.com/pytorch/xla/pull/8563

**Comment by tengyifei** [comment]
Original PR: https://github.com/pytorch/xla/pull/8524
Reason: low-risk perf optimization for the experimental scan feature
Backport PR link: https://github.com/pytorch/xla/pull/8572

**Comment by tengyifei** [comment]
Original PR: https://github.com/pytorch/xla/pull/8576
Reason: fix broken r2-6-0-rc5-libtpu-3-10-tpuvm build (http://shortn/_U3q858o5GH)
Backport PR link: https://github.com/pytorch/xla/pull/8580

**Comment by tengyifei** [comment]
Original PR: https://github.com/pytorch/xla/pull/8529
Reason: fix scan crash when an output aliases an input
Backport PR link: https://github.com/pytorch/xla/pull/8581

**Comment by rpsilva-aws** [comment]
Original PR: https://github.com/pytorch/xla/pull/8571
Reason: Allow customers to implicitly downcast to INT32, and make Neuron an opt-in.
Backport PR link: Pending merge

**Comment by rpsilva-aws** [comment]
> Original PR: [#8571](https://github.com/pytorch/xla/pull/8571) Reason: Allow customers to implicitly downcast to INT32, and make Neuron an opt-in. Backport PR link: Pending approvals

Backport PR: https://github.com/pytorch/xla/pull/8589

**Comment by tengyifei** [comment]
Original PR: https://github.com/pytorch/xla/pull/8530
Reason: fix scan crash when output dtype is not float32
Backport PR link: https://github.com/pytorch/xla/pull/8604

**Comment by tengyifei** [comment]
Original PR: https://github.com/pytorch/xla/pull/8555
Reason: without this, scan is unusable under debugging env vars
Backport PR link: https://github.com/pytorch/xla/pull/8605

**Comment by rpsilva-aws** [comment]
Original PR: https://github.com/pytorch/xla/pull/8584
Reason: Required for Neuron - safe and well-scoped interface for gradient accumulation using XLA's while loop.
Backport PR link: Pending mainline merge

**Comment by tengyifei** [comment]
Original PR: https://github.com/pytorch/xla/pull/8562
Reason: low-risk perf optimization for the experimental scan feature
Backport PR link: https://github.com/pytorch/xla/pull/8611

**Comment by zpcore** [comment]
Original PR: https://github.com/pytorch/xla/pull/8609
Reason: Need for multipod run with libtpu temporary hack.
Backport PR link:  https://github.com/pytorch/xla/pull/8618

**Comment by jeffhataws** [comment]
Original PR: https://github.com/pytorch/xla/pull/8622
Reason: In earlier torch-xla versions nprocs != 1 or None was ignored. In 2.6 we make it a value error. However the message was not clear. This change improve the messaging to be more clear and actionable.  It would be good to cherry-pick, as customers will start to see errors when nprocs not equal to 1 or None (whereas before 2.6 there was a warning only).
Backport PR link: https://github.com/pytorch/xla/pull/8623

**Comment by tengyifei** [comment]
Original PR: https://github.com/pytorch/xla/pull/8628
Reason: Necessary to fix CI tests on TPU v5 and v6
Backport PR link: https://github.com/pytorch/xla/pull/8634
